src/plugins/discover/public/application/main/utils/get_fetch_observable.ts

```diff
@@ -60,13 +60,4 @@ export function getFetch$({
     data.query.queryString.getUpdates$(),
     searchSessionManager.newSearchSessionIdFromURL$.pipe(filter((sessionId) => !!sessionId))
   ).pipe(debounceTime(100));
-
-  /**
-   * Skip initial fetch when discover:searchOnPageLoad is disabled.
-   */
-  if (initialFetchStatus === FetchStatus.UNINITIALIZED) {
-    fetch$ = fetch$.pipe(skip(1));
-  }
```

Comment:
Removed and made more implicit, not to skip a fetch in the given condition, but to fetch when it's needed and not to initially fetch when `discover:searchOnPageLoad is disabled`